components/brave_wallet_ui/common/hardware/types.ts

```diff
@@ -35,60 +33,178 @@ export type SignHardwareOperationResult = HardwareOperationResult & {
+  derivationScheme: DerivationScheme
+  coin: BraveWallet.CoinType
+  keyringId: BraveWallet.KeyringId
+  vendor: BraveWallet.HardwareVendor
+  name: string
+  /** Some schemes are valid only for a single network for corresponding
+   * coin. */
+  fixedNetwork?: string
+  /** For some schemes there is no indexing in template and only one account is
+   * supported. */
+  singleAccount?: boolean
+  pathTemplate: (index: number | 'x') => string
+}
+
+export const EthLedgerLiveHardwareImportScheme: HardwareImportScheme = {
+  derivationScheme: DerivationSchemes.EthLedgerLive,
+  coin: BraveWallet.CoinType.ETH,
+  keyringId: BraveWallet.KeyringId.kDefault,
+  vendor: BraveWallet.HardwareVendor.kLedger,
+  name: 'Ledger Live',
+  pathTemplate: (index) => `m/44'/60'/${index}'/0/0`
+}
+
+export const EthLedgerLegacyHardwareImportScheme: HardwareImportScheme = {
+  derivationScheme: DerivationSchemes.EthLedgerLegacy,
+  coin: BraveWallet.CoinType.ETH,
+  keyringId: BraveWallet.KeyringId.kDefault,
+  vendor: BraveWallet.HardwareVendor.kLedger,
+  name: 'Legacy (MEW/MyCrypto)',
+  pathTemplate: (index) => `m/44'/60'/0'/${index}`
+}
+
+export const EthLedgerDeprecatedHardwareImportScheme: HardwareImportScheme = {
+  derivationScheme: DerivationSchemes.EthLedgerDeprecated,
+  coin: BraveWallet.CoinType.ETH,
+  keyringId: BraveWallet.KeyringId.kDefault,
+  vendor: BraveWallet.HardwareVendor.kLedger,
+  name: 'Deprecated (Not recommended)',
+  pathTemplate: (index) => `m/44'/60'/${index}'/0`
+}
+
+export const SolLedgerDefaultHardwareImportScheme: HardwareImportScheme = {
+  derivationScheme: DerivationSchemes.SolLedgerDefault,
+  coin: BraveWallet.CoinType.SOL,
+  keyringId: BraveWallet.KeyringId.kSolana,
+  vendor: BraveWallet.HardwareVendor.kLedger,
+  name: 'Default',
+  pathTemplate: (index) => `44'/501'/${index}'/0'`
+}
+
+export const SolLedgerLiveHardwareImportScheme: HardwareImportScheme = {
+  derivationScheme: DerivationSchemes.SolLedgerLedgerLive,
+  coin: BraveWallet.CoinType.SOL,
+  keyringId: BraveWallet.KeyringId.kSolana,
+  vendor: BraveWallet.HardwareVendor.kLedger,
+  name: 'Ledger Live',
+  pathTemplate: (index) => `44'/501'/${index}'`
 }
 
-export enum TrezorDerivationPaths {
-  Default = 'trezor'
+export const SolLedgerBip44RootHardwareImportScheme: HardwareImportScheme = {
+  derivationScheme: DerivationSchemes.SolLedgerBip44Root,
+  coin: BraveWallet.CoinType.SOL,
+  keyringId: BraveWallet.KeyringId.kSolana,
+  vendor: BraveWallet.HardwareVendor.kLedger,
+  name: 'Bip44 Root',
+  singleAccount: true,
+  pathTemplate: (index) => `44'/501'`
 }
 
-export enum SolDerivationPaths {
-  Default = 'default',
-  LedgerLive = 'ledger-live',
-  Bip44Root = 'bip-44-root'
+export const FilLedgerMainnetHardwareImportScheme: HardwareImportScheme = {
+  derivationScheme: DerivationSchemes.FilLedgerMainnet,
+  coin: BraveWallet.CoinType.FIL,
+  keyringId: BraveWallet.KeyringId.kFilecoin,
+  vendor: BraveWallet.HardwareVendor.kLedger,
+  name: '',
```

Comment:
Should we name this?

Reply:
Changed empty names to 'Default'. However none of them could be visible for user in current implementation.
But I think we should always show dervivation path dropdown even when there is only one default scheme available just to let user know what exactly is being imported from device